scripts/templates/valddi.cpp.mako

```diff
@@ -74,37 +76,35 @@ namespace ur_validation_layer
 
         ${x}_result_t result = ${th.make_pfn_name(n, tags, obj)}( ${", ".join(th.make_param_lines(n, tags, obj, format=["name"]))} );
 
-        %if func_name == n + "AdapterRelease":
-        if( context.enableLeakChecking && result == UR_RESULT_SUCCESS )
-        {
-            refCountContext.decrementRefCount(${object_param}, true);
-        }
-        %elif func_name == n + "AdapterRetain":
+        %for tp in tracked_params:
+        <%
+            tp_handle_funcs = next((hf for hf in handle_create_get_retain_release_funcs if th.subt(n, tags, tp['type']) in [hf['handle'], hf['handle'] + "*"]), None)
+            is_handle_to_adapter = ("_adapter_handle_t" in tp['type'])
+        %>
+        %if func_name in tp_handle_funcs['create']:
         if( context.enableLeakChecking && result == UR_RESULT_SUCCESS )
         {
-            refCountContext.incrementRefCount(${object_param}, true);
+            refCountContext.createRefCount(*${tp['name']});
         }
-        %elif func_name == n + "AdapterGet":
-        if( context.enableLeakChecking && phAdapters && result == UR_RESULT_SUCCESS )
+        %elif func_name in tp_handle_funcs['get']:
+        if( context.enableLeakChecking && ${tp['name']} && result == UR_RESULT_SUCCESS )
```

Comment:
Can the `name` be an empty field?

Reply:
It should never be an empty field unless specification was incorrectly formulated.

Reply:
This check `tp['name']` can be removed then.

Reply:
This is not a check, I'm retrieving the value at `tp['name']` with `${...}`.